reactive/kubernetes_master.py

```diff
@@ -177,7 +206,9 @@ def service_cidr():
 def freeze_service_cidr():
     ''' Freeze the service CIDR. Once the apiserver has started, we can no
     longer safely change this value. '''
-    db.set('kubernetes-master.service-cidr', service_cidr())
+    if is_service_cidr_expansion():
+        db.set('kubernetes-master.service-cidr', hookenv.config(
+            'service-cidr'))
```

Comment:
```suggestion
    db.set('kubernetes-master.service-cidr', hookenv.config('service-cidr'))
```

Suggest moving the check outside the function to allow for caching the expansion check result, see below.

Reply:
This change will make it necessary to edit or remove the call to freeze_service_cidr here: https://github.com/charmed-kubernetes/charm-kubernetes-master/blob/d89d04d3b0a2c12fe7f676a5bdd130334a442f27/reactive/kubernetes_master.py#L965

Reply:
Good point.  And it's not actually necessary to move the check outside of the freeze function, so it's probably better to skip this change as well as adding the `if` around the other call.  I'll update / remove the suggested changes to avoid confusion.

Reply:
```suggestion
    frozen_service_cidr = db.get('kubernetes-master.service-cidr')
    if not frozen_service_cidr or is_service_cidr_expansion():
        db.set('kubernetes-master.service-cidr', hookenv.config(
            'service-cidr'))
```

Reply:
Modified as per your suggestion. New check added to return False if old and new service cidr is same.